lib/msal-node/src/config/Configuration.ts

```diff
@@ -30,7 +31,7 @@ export type NodeAuthOptions = {
     clientId: string;
     authority?: string;
     clientSecret?: string;
-    clientAssertion?:string;
+    clientAssertion?: ClientAssertionCredential;
```

Comment:
This is a breaking change

Reply:
Do you think that's the right way to go? If so I'll make that change, I'm not sure if there's any other alternative really. Although it's a breaking change, I don't believe anyone is using this (the client assertion flow simply shouldn't work as the code is right now).

Reply:
How do we know if the existing functionality isn't working and we aren't going to break someone?

Reply:
A unit test should be able to answer that, in theory.

Reply:
It seems a bit more complicated than I originally realized (and there are more bugs). There's a few different types of client assertions that are built on the node side which the common client class doesn't support. None of the currently implemented assertion configs in Node should work at all. 

I'll spend the day fixing this and I really hope I can get it merged by Monday.